### ISSUE TITLE
JSpecifyExperimental: false positive for addAll with ? element type

### ISSUE THREAD

**Issue by unv-unv** [uber/NullAway#1822]
JSpecifyExperimental mode on.

```java
@NullMarked
class Test {
    public Collection<Object> foo(List<?> list) {
        List<Object> result = new ArrayList<>();
        // BUG: List<?> cannot be converted to Collection<? extends Object> (List<?> is a subtype of Collection<?>)
        result.addAll(list);
        return result;
    }
}
```
Shows compilation error:
```
[ERROR]Test.java[6,22]: [NullAway] incompatible types: List<?> cannot be converted to Collection<? extends Object> (List<?> is a subtype of Collection<?>)
```
NullAway 0.14.1, Error Prone 2.50.0.

**Comment by msridhar** [comment]
The error message is bad on this one, but I think it's a real issue, since `List<?>` could contain null values. That's because its type variable has a nullable upper bound. Does that make sense? Sorry can provide links later if needed, I'm on my phone

**Comment by unv-unv** [comment]
But I have `@NullMarked` annotation. So `List<?>` should be non-nullable.

**Comment by msridhar** [comment]
See https://jspecify.dev/docs/user-guide/#wildcard-bounds.  For unbounded wildcards in `@NullMarked` code, the default upper bound is the same as the upper bound of the corresponding type variable, which in this case is `@Nullable Object`, taking into account the JSpecify JDK model of `List`.

**Comment by unv-unv** [comment]
Is it possible to specify non-null question mark?

**Comment by msridhar** [comment]
Yes! Write `List<? extends Object>`. Object here is implicitly non-null
